extensions/amp-story/1.0/amp-story-quiz.js

```diff
@@ -0,0 +1,182 @@
+      prompt.tagName[prompt.tagName.length - 1] > 3
+    ) {
+      handleError(
+        'The first child must be a heading element <h1>, <h2>, or <h3>'
+      );
+    }
+
+    prompt.setAttribute('class', 'i-amp-story-quiz-prompt');
+
+    const options = Array.from(this.element.querySelectorAll('option'));
+    if (options.length < 2 || options.length > 4) {
+      handleError('Improper number of options');
+    }
+
+    this.shadowRoot_
+      .querySelector('.i-amp-story-quiz-prompt-container')
+      .appendChild(prompt);
+
+    options.forEach((option, index) => this.configureOption_(option, index));
+
+    if (this.element.children.length !== 0) {
+      handleError('Too many children');
+    }
+  }
+
+  /**
+   * @private
+   * Creates an option container with option content,
+   * adds styling and answer choices,
+   * and adds it to the shadow DOM.
+   * @param {HTMLOptionElement} option
+   * @param {number} index
+   */
+  configureOption_(option, index) {
+    // Transfer the option information into a span -
+    // this allows the option container to house other markup,
+    // such as the answer choices
+    const convertedOption = document.createElement('span');
+    convertedOption.textContent = option.textContent;
+    if (option.hasAttribute('correct')) {
+      convertedOption.setAttribute('correct', 'correct');
+    }
+    option.remove();
+
+    // Set up the class and add it to the shadow root
+    convertedOption.setAttribute('class', 'i-amp-story-quiz-option');
+    this.shadowRoot_
+      .querySelector('.i-amp-story-quiz-option-container')
+      .appendChild(convertedOption);
+
+    // Create a container for the answer choice and add it to the shadow root
+    const answerChoice = document.createElement('span');
+    answerChoice.textContent = answerChoiceOptions[index];
+    answerChoice.setAttribute('class', 'i-amp-story-quiz-answer-choice');
+    convertedOption.prepend(answerChoice);
+  }
+
+  /**
+   * @private
+   * Attaches functions to each option to handle state transition.
+   *
+   */
+  attachOptionActionHandlers_() {
+    const options = Array.from(
+      this.shadowRoot_.querySelectorAll('.i-amp-story-quiz-option')
+    );
+
+    // attach listeners
+    options.forEach(option => {
+      option.addEventListener('click', () => {
+        if (!this.hasReceivedResponse_) {
+          options.forEach(o => {
+            o === option
+              ? o.setAttribute(
+                  'class',
+                  `i-amp-story-quiz-option i-amp-story-quiz-option-post-selection i-amp-story-quiz-option-selected`
+                )
+              : o.setAttribute(
+                  'class',
+                  `i-amp-story-quiz-option i-amp-story-quiz-option-post-selection`
+                );
+
+            const symbolContainer = o.querySelector(
+              '.i-amp-story-quiz-answer-choice'
+            );
+            // TODO: REPLACE THESE WITH ICONS
+            if (o.hasAttribute('correct')) {
```

Comment:
We should think about adding an attribute/class here, which would trigger some CSS. This way, the logic and the display are decoupled, and it'll be much easier to write CSS animations.

Reply:
Mentioned in the above comment about AMP actions, let's sync more tomorrow about the best way to go about this

Reply:
Done